spec/features/users/sign_up_spec.rb

```diff
@@ -23,6 +23,21 @@
     end
   end
 
+  context 'user cancels on the enter password screen' do
+    it 'deletes the user record and returns them to the home page' do
+      email = 'test@test.com'
+
+      visit sign_up_email_path
+
+      submit_form_with_valid_email
+      click_confirmation_link_in_email(email)
+      click_on t('links.cancel_account_creation')
+
+      expect(current_path).to eq root_path
+      expect(User.find_with_email(email)).to be_nil
```

Comment:
Why is it not acceptable to look in the DB in feature specs? I treat feature specs as integration tests that perform end to end tests, including testing things that are not visible to the user. In this case, the only way to know for sure that the user was deleted is to check the DB. If we only checked for the flash message that says "The record of your information has been cleared", that wouldn't guarantee that the user was actually deleted. Since this is testing canceling account creation, where the user expects their account to be deleted, it is critical to make sure it does indeed get deleted.